WrapImage/nifti_wrapper.py

```diff
@@ -1,11 +1,51 @@
 import argparse
 import json
 import os
+from pathlib import Path
+import subprocess
 import nibabel as nib
 from src.wrappers.OsipiBase import OsipiBase
 import numpy as np
 from tqdm import tqdm
 
+def dicom_to_niix(vol_dir: Path):
+    """
+    For converting DICOM images to a (compresssed) 4d nifti image
+    """
+
+    try:
+        res = subprocess.run(
+            [
+                "dcm2niix",
+                "-f", "%s_%p", # dcm2niix attempts to provide a sensible file naming scheme
+                "-o", vol_dir,
+                "-z", "y", #specifying compressed nii.gz file
+                # https://www.nitrc.org/plugins/mwiki/index.php/dcm2nii:MainPage
+                # for further configuration for general usage see page above
+                vol_dir         
+            ],
+            capture_output=True,
+            text=True,
+            check=True
+        )
+        
+        nifti_files = list(Path(vol_dir).glob("*.nii.gz"))
+        if not nifti_files:
+            raise RuntimeError("No NIfTI file generated by dcm2niix.")
+        if len(nifti_files) > 1:
+            raise RuntimeError("Multiple NIfTI files generated; expected one 4D NIfTI.")
+
+        bval_files = list(vol_dir.glob("*.bval"))
+        bvec_files = list(vol_dir.glob("*.bvec"))
+        bval_path = str(bval_files[0]) if bval_files else None
+        bvec_path = str(bvec_files[0]) if bvec_files else None
+
+
+        return vol_dir / "dicom2nifti.nii.gz", bval_path, bvec_path
```

Comment:
It looks like the nifti name is hard coded. Shouldn't it be specifically the file that's written this run? For example, if this is run repeatedly what happens?